bundles/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/message/PowermaxStatusMessage.java

```diff
@@ -24,10 +28,34 @@
  */
 public class PowermaxStatusMessage extends PowermaxBaseMessage {
 
-    private static final String[] EVENT_TYPE_TABLE = new String[] { "None", "Tamper Alarm", "Tamper Restore", "Open",
-            "Closed", "Violated (Motion)", "Panic Alarm", "RF Jamming", "Tamper Open", "Communication Failure",
-            "Line Failure", "Fuse", "Not Active", "Low Battery", "AC Failure", "Fire Alarm", "Emergency",
-            "Siren Tamper", "Siren Tamper Restore", "Siren Low Battery", "Siren AC Fail" };
+    private static byte[] zoneBytes(byte zones1, byte zones9, byte zones17, byte zones25) {
+        return new byte[] { zones25, zones17, zones9, zones1 };
+    }
+
+    private static boolean[] zoneBits(byte[] zoneBytes) {
+        boolean[] zones = new boolean[32];
+        char[] binary = new BigInteger(zoneBytes).toString(2).toCharArray();
+        int len = binary.length - 1;
+
+        for (int i = len; i >= 0; i--) {
+            zones[len - i + 1] = (binary[i] == '1');
+        }
+
+        return zones;
+    }
+
+    private static String zoneList(byte[] zoneBytes) {
+        boolean[] zones = zoneBits(zoneBytes);
+        List<String> names = new ArrayList<>();
+
+        for (int i = 1; i < zones.length; i++) {
+            if (zones[i]) {
+                names.add(String.format("Zone %d", i));
+            }
+        }
```

Comment:
I don't know if the result is the same as before.
I hope we have seriously tested this code.

Reply:
Yeah, this was my attempt to simplify the copy-and-paste instances of bit math. It's wordy but personally I find this approach (array of booleans) more readable than the previous:

```
            int zoneBypass = (message[8] & 0x000000FF) | ((message[9] << 8) & 0x0000FF00)
                    | ((message[10] << 16) & 0x00FF0000) | ((message[11] << 24) & 0xFF000000);

            for (int i = 1; i <= panelSettings.getNbZones(); i++) {
                updatedState.setSensorBypassed(i, ((zoneBypass >> (i - 1)) & 0x1) > 0);
            }
```

I've tested this pretty extensively and the result is the same. I'm open to suggestions though.